parsl/dataflow/task_status_poller.py

```diff
@@ -0,0 +1,62 @@
+import logging
+import parsl
+import time
+from typing import Dict, List, Sequence
+
+from parsl.dataflow.executor_status import ExecutorStatus
+from parsl.dataflow.strategy import Strategy
+from parsl.executors.base import ParslExecutor
+from parsl.providers.provider_base import JobStatus
+
+logger = logging.getLogger(__name__)
+
+
+class PollItem(ExecutorStatus):
+    def __init__(self, executor: ParslExecutor):
+        self._executor = executor
+        self._interval = executor.status_polling_interval
+        self._last_poll_time = 0.0
+        self._status: Dict[object, JobStatus] = {}
```

Comment:
This type annotation syntax is only in python >=3.6. Use  something like `# type: Dict[object, JobStatus]` on the end of the line instead

Reply:
Thanks. Wasn't 3.6 a minimum requirement for running parsl?

Reply:
Just asking. It's a straightforward change that I can do once this call I'm in ends (or maybe even before).

Reply:
3.5 is the minimum. I just made an issue about pushing for 3.6+ requirement, but that will take time.